### PR TITLE
A way to inspect the properties of an object

### PR DESCRIPTION
**Pull request by ghost**
Use like this:

```swift
struct Person {
  let name: String
}

defaultInspection(of: Person(name: "Tom")) // uses Mirror

extension Person: Inspectable {
  func configure(using setter: KeyValueSetter<Person>) {
    setter.setValue(name, forKey: "myCustomKey")
  }
}

let newInspection: Inspected<Person> = inspect(Person(name: "Tom")) // uses custom configuration
```

### CODE REVIEW

File: Sources/Inspectable.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+public final class Inspected<Object> {
+  /// The name of the stored object's type
+  public static let name: String
```
Comment (tattn) [inline Sources/Inspectable.swift]: Is this typo?

```swift
public static var name: String = ""
```

File: Sources/Inspectable.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    return value(forKey: key, asType: Data.self)
+  }
+  
+  private var dict = [String:Any]()
+  init(_instance: Object) {
+    self.name = String(describing: Object.self)
+    let mirror = Mirror(reflecting: _instance)
+    for child in mirror.children {
+      if let key = child.label {
+        dict[key].= child.value
```
Comment (tattn) [inline Sources/Inspectable.swift]: typo?

File: Sources/Inspectable.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  public func url(forKey key: String) -> URL? {
+    return value(forKey: key, asType: URL.self)
+  }
+  public func data(forKey key: String) -> Data? {
+    return value(forKey: key, asType: Data.self)
+  }
+  
+  private var dict = [String:Any]()
+  init(_instance: Object) {
+    self.name = String(describing: Object.self)
```
Comment (tattn) [inline Sources/Inspectable.swift]: This line is probably a build error.

**Comment by tattn** [comment]
Thanks for your PR 👍 
I would be grateful if you would fix code to remove build errors.

File: Sources/Inspectable.swift (hunk 1 above)
Comment (SafelySwift) [reply Sources/Inspectable.swift]: Yes is should be:

```swift
public static var name: String {
  return // etc
}
```

File: Sources/Inspectable.swift (hunk 2 above)
Comment (SafelySwift) [reply Sources/Inspectable.swift]: Fixed

**Comment by tattn** [comment]
Please add `Inspectable.swift` to the project.

**Comment by SafelySwift** [comment]
> Please add `Inspectable.swift` to the project.

I have already done so. What is your request?

**Comment by tattn** [comment]
You need to edit `./SwiftExtensions.xcodeproj/project.pbxproj` using Xcode.
`Inspectable.swift` was added to this branch, but now it is not added to the Xcode project.

**Comment by SafelySwift** [comment]
I am not using GitHub desktop for this, should I use it? I do not fully understand the issue. I have pushed  to master but it is not there?
*(Sorry, I am new to GitHub, so forgive me if this is basic knowledge)*

**Comment by SafelySwift** [comment]
screen shot 2019-01-08 at 11 02 12 pm

**Comment by tattn** [comment]
Please try the following steps:

1. Clone your forked repository (https://github.com/SafelySwift/SwiftExtensions)
2. Change the branch to `patch-2`
3. Open a project (`SwiftExtensions.xcodeproj`) in the cloned repository using Xcode.app
4. Drag & Drop `Inspectable.swift` to `Sources` in the project navigator
image
5. Build on Xcode.app
6. Add & commit your changes
7. Push

I don't know good English references for GitHub... (Sorry, I'm Japanese...)

**Comment by SafelySwift** [comment]
I understand. Thanks for the explanation.

**Pull request closed without merging**